### PR TITLE
Add ephemeral iOS directory to put future generated files

### PR DESCRIPTION
**Pull request by jmagman**
Future prevention of https://github.com/flutter/flutter/issues/39491 on iOS.  There's nothing currently generated in this ephemeral directory, but I'm about to put out a PR that will generate a few files to this location.

- Rename iOS module directory property `ephemeralDirectory` -> `ephemeralModuleDirectory`
- Create a project property for `Flutter/ephemeral` directory.  
- Remove the `ephemeralDirectory` on `flutter clean`
- Add to .gitignore

This doesn't actually move any existing generated files to this location--that can be done in individual follow-up PRs as needed.

### CODE REVIEW

File: packages/flutter_tools/templates/app/ios.tmpl/.gitignore [hunk 1]
```diff
@@ -18,6 +18,7 @@ Flutter/App.framework
 Flutter/Flutter.framework
 Flutter/Flutter.podspec
 Flutter/Generated.xcconfig
+Flutter/ephemeral/
```
Comment (jmagman) [inline packages/flutter_tools/templates/app/ios.tmpl/.gitignore:21]: \cc @RedBrogdon Sorry here's another directory to add to all the iOS sample `.gitignore`s.  But new ones should be rare in the future with this PR.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
> Sorry here's another directory to add to all the iOS sample `.gitignore`s. But new ones should be rare in the future with this PR.

Oh that reminds me, better do the same for the examples/integration_tests in this repo...

File: .gitignore [hunk 2]
```diff
@@ -92,6 +92,7 @@ unlinked_spec.ds
 **/ios/Flutter/Flutter.framework
 **/ios/Flutter/Flutter.podspec
 **/ios/Flutter/Generated.xcconfig
+**/ios/Flutter/ephemeral
```
Comment (jmagman) [inline .gitignore:95]: The other non-template `.gitignore`s aren't necessary on top of this one, but let's be consistent.

**Review by stuartmorgan-g — APPROVED** [review]
Nice, LGTM!

Apologies for the naming collision; I didn't realize there was already something called "the ephemeral directory" when I made this concept for desktop, and by the time I realized it would have been a giant pain to change.

**Comment by jmagman** [comment]
> Apologies for the naming collision

Naming is hard.  "modules" also references several concepts.  😄

**Pull request merged**